hicexplorer/hicPlotViewpoint.py

```diff
@@ -177,7 +184,71 @@ def main(args=None):
+                        log.info('start {}, end {} , interactions[start:end] {}'.format(start, end, interactions[start:end]))
+                    average_contacts_[i] = np.mean(interactions[start:end])
+                # handle border conditions
+                for i in range(window_size):
+                    start = i - window_size_upstream
+                    if start < 0:
+                        start = 0
+                    end = i + window_size + 1
+                    # log.info('start {}, end {}'.format(start, end))
+                    # log.info('interactions[start:end]: {}'.format(interactions[start:end]))
+
+                    # log.info('interactions[-end:]: {}'.format(interactions[-end:]))
+
+                    average_contacts_[i] = np.mean(interactions[start:end])
+                    average_contacts_[-(i + 1)] = np.mean(interactions[-end:])
+                average_contacts.append(average_contacts_)
+                log.info(average_contacts_)
+
+            for i, interactions_list_ in enumerate(interactions_list):
+                with open(args.interactionOutFileName + '_' + matrix_name_legend[i] + '.bedgraph', 'w') as fh:
+                    for j, interaction in enumerate(interactions_list_):
+                        fh.write("{}\t{}\t{}\t{}\t{}\t{}\t{:.12f}\t{:.12f}\n".
+                                 format(toString(interaction[0]), toString(interaction[1]), toString(interaction[2]),
+                                        toString(interaction[3]), toString(interaction[4]),
+                                        toString(interaction[5]), float(interaction[6]),
+                                        float(average_contacts[i][j])))
+
+                        # print("{}\t{}\t{}\t{}\t{}\t{}\t{:.12f}\t{:.12f}\n".\
+                        # format(toString(interaction[0]), toString(interaction[1]), toString(interaction[2]),
+                        #         toString(interaction[3]), toString(interaction[4]),
+                        #         toString(interaction[5]), float(interaction[6]),
+                        #         float(average_contacts[i][j])))
+                return
+
         for i, interactions_list_ in enumerate(interactions_list):
+            print(matrix_name_legend[i])
             with open(args.interactionOutFileName + '_' + matrix_name_legend[i] + '.bedgraph', 'w') as fh:
                 for interaction in interactions_list_:
-                    fh.write("{}\t{}\t{}\t{}\t{}\t{}\t{:.12f}\n".format(toString(interaction[0]), toString(interaction[1]), toString(interaction[2]), toString(interaction[3]), toString(interaction[4]), toString(interaction[5]), float(interaction[6])))
+                    fh.write("{}\t{}\t{}\t{}\t{}\t{}\t{:.12f}\n".
+                             format(toString(interaction[0]), toString(interaction[1]),
```

Comment:
These toSting() conversion are really costly, especially because format() is doing this for you already. Same for float()

Reply:
toString is used to prevent Python 2 / 3 issues.